parcels/field.py

```diff
@@ -152,25 +152,44 @@ def from_netcdf(cls, filenames, variable, dimensions, indices=None, grid=None,
 
         if not isinstance(filenames, Iterable) or isinstance(filenames, str):
             filenames = [filenames]
-        dimension_filename = dimension_filename if dimension_filename else filenames[0]
+
+        data_filenames = filenames['data'] if type(filenames) is dict else filenames
+        if type(filenames) == dict:
+            assert len(filenames['lon']) == 1
+            assert filenames['lon'] == filenames['lat']
```

Comment:
why can't 'lon' and 'lat' not be in different files? Seems too strict to me? Or, if we need to be this strict, then at least add an error message